src/commands/data/query.ts

```diff
@@ -81,6 +88,22 @@ export class DataSoqlQueryCommand extends SfCommand<unknown> {
     }),
     'result-format': resultFormatFlag(),
     perflog: perflogFlag,
+    'output-file': Flags.file({
+      summary: messages.getMessage('flags.output-file.summary'),
+      relationships: [
+        {
+          type: 'some',
+          flags: [
+            {
+              name: 'result-format',
+              // eslint-disable-next-line @typescript-eslint/require-await
+              when: async (flags): Promise<boolean> =>
+                flags['result-format'] === 'csv' || flags['result-format'] === 'json',
```

Comment:
validates `--output-file` is only specified if `--result-format` value is `csv` or `json`.